src/Credentials/CredentialProvider.php

```diff
@@ -75,10 +78,20 @@ public static function defaultProvider(array $config = [])
             );
         }
 
+        $defaultConfigFileName = self::getHomeDir() . '/.aws/config';
         return self::memoize(
             self::chain(
                 self::env(),
+                self::assumeRole(),
+                self::assumeRole(
+                    'profile assumes_role',
+                    $defaultConfigFileName
+                ),
                 self::ini(),
+                self::ini(
```

Comment:
This new chain setup looks really expensive -- we are potentially reading and ini parsing both the credentials files and config files twice just in this chain, and looking at the assumeRole method, that number could climb to 6.

Have you benchmarked this? How does this affect the common usage of the SDK with environment variables and instance profile credentials?